### ISSUE TITLE
Advice: merging 2 geoip pools (ipv4 pool and ipv6 pool)

### ISSUE THREAD

**Issue by extremeshok** [gdnsd/gdnsd#145]
We have 2 pools of CDN servers using geoip.
CDN4 constists of geo-located ipv4 servers.
CDN6 consists of geo-located ipv6 servers.

*not every location has a IPv4 and a IPv6, hence the 2 separate pools.

I need a way to merge the 2 pools so that every query will respond with a geolocated IPv4 (A) and a geolocated IPv6 (AAAA)

Any advice would be appreciated, and we will buy you a steak dinner if you have a working solution.

Basically the following needs to be true
````
dig @dns.local www.mydomain.com
-->
;; ANSWER SECTION:
www.mydomain.com.    300     IN      A       11.22.33.44
www.mydomain.com.    300     IN     AAAA       AA::BB::CC::DD::EE
````

**Comment by mattlehner** [comment]
I maybe just solved this:

;; QUESTION SECTION:
;ns1.example.net.               IN      A

;; ANSWER SECTION:
ns1.example.net.        120     IN      A       192.168.10.1
ns1.example.net.        120     IN      A       192.168.10.2

;; AUTHORITY SECTION:
example.net.    300     IN      NS      ns1.example.com.
example.net.    300     IN      NS      ns1.example.net.

;; ADDITIONAL SECTION:
ns1.example.net.        120     IN      AAAA    fe80::f816:3eff:fee3:48bc

In the dcmap, I did:

      dc1 => {
        addrs_v4 => [ 192.168.10.1, 192.168.10.2 ],
        addrs_v6 => fe80::f816:3eff:fee3:48bc
        },

I'm just messing around with gdnsd, and I'm not sure that's exactly what you want, but I think it's probably close.

**Comment by blblack** [comment]
I think the original problem isn't solvable in gdnsd-2.x.  What this really needs is independent mapping per-address-family for a single hostname (where IPv4 may resolve to one DC and IPv6 to another, and in general the DCs are mixed in which supports which address families).

The preliminary work on gdnsd-3.x allows for this better, because the existing `DYNA` has been split into separate `DYNA` + `DYNAAAA` which resolve address families independently (in the geoip case, that would be with completely separate resources, which may or may not share a map).

**Comment by blblack** [comment]
The preliminary work mentioned above didn't make the cut for 3.x.  There are plans to still go this direction (splitting address families fully at the DYNA level) for a future 3.N or 4.0 release, but that's not imminent.

**Comment by danhanks** [comment]
@blblack Curious if you're still thinking about this for a 3.N release?

**Comment by blblack** [comment]
No, but it will be there in 4.x I believe.  It will be more like generically supporting all the RR-types eventually, e.g. `DYN:A` vs `DYN:AAAA` vs `DYN:PTR` or whatever.
